ui/tests/unit/services/flags-test.js

```diff
@@ -66,8 +69,12 @@ module('Unit | Service | flags', function (hooks) {
       assert.deepEqual(this.service.activatedFlags, [], 'Activated flags are empty');
     });
 
-    test('it returns an empty array if the cluster is OSS', async function (assert) {
-      this.owner.lookup('service:version').type = 'community';
+    test('it does not call activation-flags endpoint if the cluster is OSS', async function (assert) {
+      this.version.type = 'community';
+
+      this.server.get('sys/activation-flags', () => {
+        assert.true(false, 'activation-flags is not called');
```

Comment:
If we don't want this called, could we have this endpoint throw an error? This can make it a little clearer that the assertion inside that mirage request should _not_ actually be hit

https://github.com/hashicorp/vault/blob/1e8eefade13acf0168a12dfbf7af4ef526f3a757/ui/tests/integration/components/dashboard/overview-test.js#L105-L108
```js
    this.server.get(
        'sys/activation-flags',
        () => new Error('uh oh! a request was made to sys/activation-flags, this should not happen for community versions')
      );

Reply:
👆 love that idea!

Reply:
@noelledaley You gave it to me on one of my recent PRs! 🎉 😄